docs/ca/dev/opentelemetry/how-to-instrument.md

```diff
@@ -0,0 +1,414 @@
+---
+title: OpenTelemetry overview
+description: Describing what is OpenTelemetry and how Spryker integrates it in the SCCOS
+template: howto-guide-template
+last_updated: Feb 1, 2025
+related:
+---
+
+You may want to get familiar with [OpenTelemetry Documentation first](https://opentelemetry.io/docs/concepts/), but we are going to give you a brief overview here too.
+
+
+### Convention following
+The current implementation is following https://opentelemetry.io/docs/specs/semconv/ version 1.30.0.
+
+### Trace
+Trace represents a single transaction. It has a unique ID and all spans are related to it. Trace has a name that is defined automatically or can be changed as described [here](https://github.com/spryker/spryker-docs/edit/otel-main-doc/docs/ca/dev/opentelemetry/how-to-instrument.md#integration).
+
+#### Span
+A span represents a unit of work or operation. Spans are the building blocks of Traces. They include the following information:
+- Name
+- Parent span ID (empty for root spans)
+- Start and End Timestamps
+- Span Context
+- Attributes
+- Span Events
+- Span Links
+- Span Status
+
+You can compare a Span to an execution of a single method.
+
+#### Hook
+Hook is a function that executes a closure before and after method execution, providing a way to instrument code without modifying it directly.
+When you execute your method, `pre` closure is executing and should open a span. After the method is executed, `post` closure is executed and should catch if method was executed successfully, add missing attributes (you can add a method return value, but we would not recommend it) and close a span.
```

Comment:
Last line in this paragraph is not clear, rewrite

Reply:
@gechetspr i think this more for application part